app/models/course.rb

```diff
@@ -322,10 +322,12 @@ def self.entry_requirement_options_without_nil_choice
   validates :sites, presence: true, on: %i[publish new]
   validates :study_sites, presence: true, on: %i[publish new], if: -> { recruitment_cycle_after_2023? }
   validates :subjects, presence: true, on: :publish
+  validates :accrediting_provider, presence: true, on: :publish
   validate :validate_enrichment_publishable, on: :publish
   validate :validate_site_statuses_publishable, on: :publish
   validate :validate_provider_visa_sponsorship_publishable, on: :publish, if: -> { recruitment_cycle_after_2021? }
   validate :validate_provider_urn_ukprn_publishable, on: :publish, if: -> { recruitment_cycle_after_2021? }
+  validate :validate_accredited_provider_is_accredited, on: :publish
```

Comment:
The behaviour of the validation works well, but I have a product question on where the link should take the user.

@MylesJarvis @Kateharries if a user tries to publish a course with an accredited provider that has lost it's accreditation, do we want to take the user to the page where they can add a completely new accredited provider, or do we want to take them to a page where they can update the course with an accredited provider that they already have? The content of the link implies to me that we are taking them to update rather than add.

<img width="1087" alt="image" src="https://github.com/DFE-Digital/publish-teacher-training/assets/50492247/cf169220-93bf-4460-b23e-f7534328b81f">

Reply:
This is my fault! I did my best in Kate's absence on Friday. But clearly, I don't know how Publish works!! I will wait for @Kateharries to advise. Sorry for causing confusion!

Reply:
`if a user tries to publish a course with an accredited provider that has lost it's accreditation`

This should never happen. It's a bug downstream. We shouldn't actually be handling this here.

Reply:
I think (for now) we should take them to the page where they need to add a new AP. Content should read: Add an accredited provider.

When they get to the new page, can we have some content on this page especially for this scenario? It could read:

[AP name] is no longer an accredited provider. You'll need to add another accredited provider and then go to the 'Basic details' tab in your course page to attach the new accredited provider to the course.